butte-build/src/codegen/tests/mod.rs

```diff
@@ -0,0 +1,78 @@
+pub use super::*;
+pub use pretty_assertions::assert_eq;
+use proc_macro2::TokenStream;
+use quote::quote;
+use std::{
+    io::Write,
+    process::{Command, Stdio},
+};
+
+// Utility functions for tests
+
+macro_rules! parse_expected_file {
+    ($file:expr) => { {
+        let src = include_str!($file);
+        let syntax = syn::parse_file(src).expect("Unable to parse file");
+        let formatted = rustfmt(&quote!(#syntax));
+
+        // Split the String per lines to make debugging/diffing
+        // viable (it also improves diff performance at test time)
+        let v : Vec<_> = formatted.lines().map(ToString::to_string).collect();
```

Comment:
Does this need to be assigned to a variable?

Reply:
`collect` needs to know the required type, so either this variable or using the turbofish operator.  I'm partial to variables because it's usually more explicit/readable, expresses the intent better.

Reply:
Sure I just mean as opposed to turbofish, which I find easier for reading an expression from start to finish without having to go back some number of lines to remember what's being collected. However I don't feel too strongly about it.

Reply:
> However I don't feel too strongly about it.

Same :slightly_smiling_face:.  My main grippe with turbofish is that sometimes functions take more than one generic, and there's more to annotate than just one type even though we're interested in only one of them: that's not the case for `collect` so I'm fine with both.